gcsweb/cmd/gcsweb/gcsweb.go

```diff
@@ -270,72 +303,73 @@ func (s *server) gcsRequest(w http.ResponseWriter, r *http.Request) {
 	path := strings.TrimPrefix(r.URL.Path, gcsPath)
 	// e.g. "/bucket/path/to/object" -> ["bucket", "path/to/object"]
 	bucket, object := splitBucketObject(path)
+	objectLogger := logger.WithFields(logrus.Fields{"bucket": bucket, "object": object})
+
+	objectLogger.Info("Processing request...")
+	// Getting the object attributes directly will determine if is a folder or a file.
+	objAttrs, _ := s.storageClient.Bucket(bucket).Object(object).Attrs(context.Background())
+
+	// This means that the object is a file.
+	if objAttrs != nil {
+		obj := s.storageClient.Bucket(bucket).Object(object)
+
+		r, err := obj.NewReader(context.Background())
```

Comment:
Please use a different variable name. `r` is already used for the `*http.Request` parameter which makes the code after this a bit confusing.